src/psyclone/tests/psyGen_test.py

```diff
@@ -638,6 +639,65 @@ def test_invokeschedule_can_be_printed():
     assert "InvokeSchedule:\n" in output
 
 
+# Loop class test
+def test_loop_create():
+    '''Test that the create method in the Loop class correctly
+    creates a Loop instance.
+
+    '''
+    loop = Loop.create("i", Literal("0"), Literal("1"), Literal("1"),
+                       [Assignment.create(Reference("tmp"), Reference("i"))])
+    result = FortranWriter().loop_node(loop)
+    assert result == "do i = 0, 1, 1\n  tmp=i\nenddo\n"
+
```

Comment:
Ah, I see the parent -> child connection is tested implicitly by the FortranWriter recusing down, I didn't catch this in the first review  but I don't think this is proper unit-testing as a bug on FortranWriter would make this test fail as well. Could you replace the assert by the explicit  `assert child in node.children` for each connection.

And more importantly the child -> parent connection is never checked.

This applies for all the `create` tests with valid parameters.

Reply:
Good point. I was relying on the writer, but as you point out a) I shouldn't and b) it does not test everything anyway.

Reply:
OK, I've done this (and found two missing child->parent links!). It's good to test.